core.go

```diff
@@ -47,82 +40,158 @@ func (c *core) Write(ent zapcore.Entry, fs []zapcore.Field) error {
 	event.Message = ent.Message
 	event.Timestamp = ent.Time
 	event.Level = sentrySeverity(ent.Level)
-	event.Platform = "Golang"
 	event.Extra = clone.fields
 	event.Tags = c.cfg.Tags
+	event.Exception = clone.createExceptions()
 
-	if !c.cfg.DisableStacktrace {
-		trace := sentry.NewStacktrace()
-		if trace != nil {
-			trace.Frames = filterFrames(trace.Frames)
-			event.Exception = []sentry.Exception{{
-				Type:       ent.Message,
-				Value:      ent.Caller.TrimmedPath(),
-				Stacktrace: trace,
-			}}
+	if event.Exception == nil && !c.cfg.DisableStacktrace && c.client.Options().AttachStacktrace {
+		stacktrace := sentry.NewStacktrace()
+		if stacktrace != nil {
+			stacktrace.Frames = filterFrames(stacktrace.Frames)
+			event.Threads = []sentry.Thread{{Stacktrace: stacktrace, Current: true}}
 		}
 	}
 
 	hub := c.cfg.Hub
 	if hub == nil {
 		hub = sentry.CurrentHub()
 	}
+
 	_ = c.client.CaptureEvent(event, nil, hub.Scope())
 
 	// We may be crashing the program, so should flush any buffered events.
 	if ent.Level > zapcore.ErrorLevel {
-		c.client.Flush(c.flushTimeout)
+		return c.Sync()
 	}
+
 	return nil
 }
 
 func (c *core) Sync() error {
 	c.client.Flush(c.flushTimeout)
+
 	return nil
 }
 
+func (c *core) createExceptions() []sentry.Exception {
+	errorsCount := len(c.errs)
+
+	if errorsCount == 0 {
+		return nil
+	}
+
+	processedErrors := make([]error, 0, errorsCount)
+	exceptions := make([]sentry.Exception, 0, errorsCount)
+
+ErrsLoop:
+	for i := errorsCount - 1; i >= 0; i-- {
+		err := c.errs[i]
+
+		for i := 0; i < maxErrorDepth && err != nil; i++ {
+			for _, processedErr := range processedErrors {
+				if err == processedErr {
```

Comment:
seems like we could use a map here

Reply:
Yeah I don't know why but thought that the slice will perform better, but I ran a quick benchmark and it isn't the case. I'll update this to a map :)